### PR TITLE
aida 3.2.1 (new formula)

### PR DESCRIPTION
**Pull request by riffard**
Aida is a hearder only cpp package that define abstract interfaces for common physics analysis objects, such as histograms, ntuples, fitters, IO etc.. The adoption of these interfaces should make it easier for physicists to use different tools without having to learn new interfaces or change all of their code. Additional benefits will be interoperability of AIDA compliant applications (for example by making it possible for applications to exchange analysis objects via XML).

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting? 
Not needed actually, this is a header only package.
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/aida-header.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class AidaHeader < Formula
+  desc "Abstract Interfaces for Data Analysis define interfaces for physics analysis"
+  homepage "https://aida.freehep.org/index.thtml"
+  url "ftp://ftp.slac.stanford.edu/software/freehep/AIDA/v3.2.1/aida-3.2.1-src.tar.gz"
+  sha256 "882d351bc09e830ae2eb512a2cbf44af5a82ef8efe31fbe0d047363da8314c81"
+  license "LGPL-3.0-or-later"
+  def install
+    mkdir_p(include.to_s)
+    cp_r("src/cpp/AIDA", "#{include}/.")
```
Comment (SMillerDev) [inline Formula/aida-header.rb:9]:
```suggestion
    include.install "src/cpp/AIDA"
```

File: Formula/aida-header.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class AidaHeader < Formula
+  desc "Abstract Interfaces for Data Analysis define interfaces for physics analysis"
+  homepage "https://aida.freehep.org/index.thtml"
+  url "ftp://ftp.slac.stanford.edu/software/freehep/AIDA/v3.2.1/aida-3.2.1-src.tar.gz"
+  sha256 "882d351bc09e830ae2eb512a2cbf44af5a82ef8efe31fbe0d047363da8314c81"
+  license "LGPL-3.0-or-later"
+  def install
+    mkdir_p(include.to_s)
```
Comment (SMillerDev) [inline Formula/aida-header.rb:8]: This isn't needed with my suggestion.

File: Formula/aida-header.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      #include <AIDA/AIDA.h>
+
+      int main() {
+        std::cout<<"AIDA version "<<AIDA_VERSION<<std::endl;
+        return 0;
+      }
+    EOS
+
+    system ENV.cxx, "test.cpp", "-I#{include}"
+    system "./a.out"
```
Comment (SMillerDev) [inline Formula/aida-header.rb:22]: Can we check some output here? Using `shell_output` and `assert_match`

File: Formula/aida-header.rb (hunk 1 above)
Comment (riffard) [reply Formula/aida-header.rb:9]: I didn't know about this feature, this is pretty useful . I applied the proposed modification.

File: Formula/aida-header.rb (hunk 2 above)
Comment (riffard) [reply Formula/aida-header.rb]: I applied the proposed modification.

File: Formula/aida-header.rb (hunk 3 above)
Comment (riffard) [reply Formula/aida-header.rb]: I added a test on the shell output.

File: Formula/aida-header.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+class AidaHeader < Formula
+  desc "Abstract Interfaces for Data Analysis define interfaces for physics analysis"
+  homepage "https://aida.freehep.org/index.thtml"
+  url "ftp://ftp.slac.stanford.edu/software/freehep/AIDA/v3.2.1/aida-3.2.1-src.tar.gz"
+  sha256 "882d351bc09e830ae2eb512a2cbf44af5a82ef8efe31fbe0d047363da8314c81"
+  license "LGPL-3.0-or-later"
+  def install
+    include.install "src/cpp/AIDA"
+  end
+  test do
```
Comment (SMillerDev) [inline Formula/aida-header.rb:10]:
```suggestion
  license "LGPL-3.0-or-later"

  def install
    include.install "src/cpp/AIDA"
  end

  test do
```

**Review by SMillerDev — APPROVED** [review]
Thanks @riffard  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**